peer/list.go

```diff
@@ -44,9 +44,6 @@ type Chooser interface {
 // A Chooser will implement the PeerChangeListener interface in order to receive
 // updates to the list of Peers it is keeping track of
 type List interface {
-	// Add a peer to the List (Called directly from a PeerProvider)
-	Add(Identifier) error
-
-	// Remove a peer from the List (Called directly from a PeerProvider)
-	Remove(Identifier) error
+	// Update performs the additions and removals to the Peer List
+	Update(additions, removals []Identifier) error
```

Comment:
I would use a `ListUpdate` struct here instead of passing `nil` for either parameter when only one operation is needed:

```
myPeerList.Update(peer.ListUpdate{Add: additions})
myPeerList.Update(peer.ListUpdate{Remove: removals})

// vs

myPeerList.Update(additions, nil)
myPeerList.Update(nil, removals)
```